### PR TITLE
Lava turf

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Ignites (max firestacks) and damages mobs (20 burn damage) that move through it. This means firesuit proofs will only partially protect you.

Ignites objects that enter it (even fireproof objects)

Slows movement of mobs trying to move through it.

Can't build on it

To do in future updates or in this PR if it sits long enough: Ways to build on lava, use WJs new/better lava icons

Let me know if you want anything tweaked Ausops

:cl: Kor
rscadd: Added a lava turf. Expect to see it in away missions or in admin "events"
/:cl:

### CODE REVIEW

**Comment by WJohn** [comment]
LAVA WORLD SLOWLY BUT SURELY.

**Comment by MMMiracles** [comment]
It'd be nice if the fireloss/firestacks was tweakable via variable for more customization.

**Comment by Ausops** [comment]
brilliant

merge pls

File: code/game/turfs/simulated/floor/plating.dm [hunk 1]
```diff
@@ -159,18 +159,38 @@
...
 
-/turf/simulated/floor/plating/abductor
-	name = "alien floor"
-	icon_state = "alienpod1"
-
-/turf/simulated/floor/plating/abductor/New()
-	..()
-	icon_state = "alienpod[rand(1,9)]"
\ No newline at end of file
+/turf/simulated/floor/plating/lava/Entered(atom/movable/AM)
```
Comment (RemieRichards) [inline code/game/turfs/simulated/floor/plating.dm]: I can enter the tile, resist, and survive like this.
When sticky made his lava tile (Which I dont know if that was ever merged...) we encouraged him to add a ```process()``` that simply called code similar to what you have below, but on the contents list of the turf.
(so long as there were contents, obviously) with some extra code to add/remove it from the processing list (```Entered()```) for further robust code.

**Comment by WJohn** [comment]
How does heat resistance or armor/augmentations protect you from lava, if at all? Are you in any way better off wading through lava while wearing a CE hardsuit than when naked? What about with full augmentations, or firesuits, or mechs? Are cyborgs affected? What about simple animals?

File: code/game/turfs/simulated/floor/plating.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/game/turfs/simulated/floor/plating.dm]: I could do that. I figured spamming resist and meds on yourself for the privilege of saying "haha I'm on lava" wouldnt be the worst thing in the world though. Back during google code there was a big push to cut down on processing objects so I am hesitant to use them (even if they're only temporary).

**Comment by KorPhaeron** [comment]
@WJohn 

It stops the fire from killing you as fast if you have heat stuff. Someone in the CE cuit could move slowly through lava so long as they brought meds.

Borgs/simple animals take the fire damage.

Mechs catch fire.

Ausops ans I figured any full immunity to lava would result in spree killers hiding in their safe zone/dragging people in

**Comment by MMMiracles** [comment]
I've found while messing around with this that there's some weird results due to setting every object to be burnable.

Laser weapons drop their internal laser lens/internal powercell, ballistics like shotguns drop their internal magazine.

Anything that results from the burning sits the lava unharmed.

Mechs turn into 1 singular ash pile (rip).

Getting teleported into a tile directly does nothing to you (rip pits dropping you straight into a lava lake).

**Comment by RemieRichards** [comment]
@MMMiracles most of that is due to the lack of a ```process()``` which I told Kor to add.

**Comment by Ausops** [comment]
I actually have one request if possible/not already done, can the lava turfs emit a small amount of light?

**Comment by KorPhaeron** [comment]
Oh duh yeah I completely forgot

**Comment by YotaXP** [comment]
I can't help but notice that this does not heat the air.

**Comment by KorPhaeron** [comment]
@MMMiracles @RemieRichards 

It should be done now. I can add a smoothlava variant that uses WJohnstons new sprites if he sends them to me

**Comment by KorPhaeron** [comment]
@YotaXP 

It doesn't produce heat because that would put a ton of strain on the server when it would eventually activate every tile on the Z level

**Comment by KorPhaeron** [comment]
@duncathan 

Could you merge this some @MMMiracles can finish their map?

**Comment by duncathan** [comment]
lg2m

**Comment by JJRcop** [comment]
https://www.youtube.com/watch?v=uh4dTLJ9q9o

**Pull request merged**